portal/server/controllers/utilisation-report-service/previous-reports/index.js

```diff
@@ -0,0 +1,49 @@
+const utilisationData = require('./data.json');
+
+const getMonthName = (monthNumber) => {
+  const date = new Date();
+  // offset by 1 as January = 1 in Database
+  date.setMonth(monthNumber - 1);
+  return date.toLocaleString('default', { month: 'long' });
+};
+
+const getUtilisationReportDownload = async (req, res) => {
+  // get data from the DB (currently using json) - will need to pass in bank id from user
+  try {
+    const { targetYear } = req.query;
+    const navItems = utilisationData.map((utilisation) => ({
+      text: utilisation.year,
+      href: `?targetYear=${utilisation.year}`,
+      attributes: { 'data-cy': `side-navigation-${utilisation.year}` },
+    }));
+
+    const utilisation = targetYear ? utilisationData.find((x) => x.year.toString() === targetYear) : utilisationData[0];
+    const utilisationIndex = targetYear ? utilisationData.findIndex((x) => x.year.toString() === targetYear) : 0;
+    if (navItems.length) {
+      navItems[utilisationIndex].active = true;
+    }
+
+    const reports = navItems.length
```

Comment:
just to be safe let's chuck a load of optional chaining in here and the below lines e.g. navItems?.length and utilisation?.reports?.map.
When we get this from the API we'll have to think about error cases and if we always have an empty array etc. being returned so a few of these pieces of logic don't fail if an API call fails